### PR TITLE
Explicitly convert gate params to float in to_matrix

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Explicitly convert the parameters of the 2-q rotations to floats to ensure that bound parameters are converted to floats and allow calling e.g. trigonometric functions. This has already been done for RZZ, RXX but RYY and RZX have been missed.

### Details and comments

```python
from qiskit.circuit import QuantumCircuit, Parameter
from qiskit.circuit.library import ExcitationPreserving, RYYGate
from qiskit.quantum_info import Statevector

p = Parameter('p')

circuit = QuantumCircuit(2)
circuit.ryy(p, 0, 1)
bound = circuit.bind_parameters({p: 1})

s = Statevector.from_label('00')
s.evolve(bound)
```
yields
```
TypeError: loop of ufunc does not support argument 0 of type ParameterExpression which has no callable cos method
```

### CODE REVIEW

**Comment by Cryoris** [comment]
It can't be cast at init time since it might be a `ParameterExpression`, but when we convert it to a matrix we know that it must be a float

**Comment by 1ucian0** [comment]
I followed a bit this rabbit hole for too long. Here are my alternative suggestions:
 - `ParameterExpression` should be extended to support the operations used in `to_matrix`, see alternative PR in https://github.com/Qiskit/qiskit-terra/pull/4919
 - Once a parameter does not have free variables, it can be converted to a native type. This allows to consider this parameter as any other parameter, see alternative PR in https://github.com/Qiskit/qiskit-terra/pull/4920

I think @kdk probably have an opinion here.

**Comment by 1ucian0** [comment]
After talking with @kdk , it seems that this cast is the expected way. Closing my PRs in that case :(

File: test/python/circuit/test_parameters.py [hunk 1]
```diff
@@ -1192,6 +1194,29 @@ def test_conjugate(self):
...
+        gate = gate_class(*params)
+        num_qubits = gate.num_qubits
+
+        circuit = QuantumCircuit(num_qubits)
+        circuit.append(gate_class(*free_params), list(range(num_qubits)))
+        bound_circuit = circuit.assign_parameters({free_params: params})
+
+        numpy.testing.assert_array_almost_equal(Operator(bound_circuit).data, gate.to_matrix())
+
+        # (Operator(bound_circuit).equal(gate.to_matrix()))
```
Comment (1ucian0) [inline test/python/circuit/test_parameters.py:1218]:
```suggestion
```

**Pull request merged**